src/shared/utils/hooks/usePasswordInput.ts

```diff
@@ -0,0 +1,27 @@
+import { useMemo, useState } from "react"
+
+import { Eye, EyeSlash, Icon } from "react-bootstrap-icons"
+
+type PasswordInputType = "text" | "password"
+
+export const usePasswordInput = (): [PasswordInputType, () => void, Icon] => {
```

Comment:
This hook has two responsibilities instead of one: 
1. it manages the password type.
2. it manages what type of Icon should be returned.

Please leave out the icon handling from here. Instead, create a separate PasswordIcon component and manage the icon through props. For example, a prop `isSecured` === true will return `<EyeSlash />` else `<Eye />`

Regarding this hook, please rename it to usePasswordType.

Reply:
Done